lib/spack/docs/packaging_guide.rst

```diff
@@ -1508,10 +1508,7 @@ Additional hybrid dependency types are (note the lack of quotes):
   * **<not specified>**: ``type`` assumed to be ``("build",
     "link")``. This is the common case for compiled language usage.
   * **alldeps**: All dependency types.  **Note:** No quotes here
```

Comment:
`alldeps` can probably be eliminated from the user docs and packages, as well.  It's used internally by the Spec traversal routines, which need a way to guarantee that they are going over the *whole* DAG.  

Using `alldeps` in a package is going be very surprising when we add things like `test` dependencies.  There are 4 packages that currently use `alldeps` -- those should probably be changed to `('build', 'link', 'run')`.